### ISSUE TITLE
[Cosmetic filters] Improve performance of tracking DOM mutations (including subframes) 

### ISSUE THREAD

**Issue by atuchin-m** [brave/brave-browser#25614]
We found that in some cases the performance of a mutation observer to track DOM changes is far from acceptable.
Particularly, this problem results in degrading speedometer2 score when cosmetic filters are enabled in subframes.

We need alternative strategies to fix the performance in such cases. 

Slack thread: https://bravesoftware.slack.com/archives/C3T9S9WKD/p1662544398836149

**Comment by LaurenWags** [comment]
@atuchin-m please add a test plan for QA, marking as `QA/Blocked` until we have that, thanks!

cc @rebron @kjozwiak

**Comment by srirambv** [comment]
Discussed with @atuchin-m and this one requires another uplift discussed [here](https://bravesoftware.slack.com/archives/CHGKGMHDJ/p1662678056036739).

**Comment by kjozwiak** [comment]
> Discussed with @atuchin-m and this one requires another uplift discussed [here](https://bravesoftware.slack.com/archives/CHGKGMHDJ/p1662678056036739).

yup, needs be verified via https://github.com/brave/brave-variations/pull/360 on `staging` before we can uplift https://github.com/brave/brave-variations/pull/361 into `production`. Once https://github.com/brave/brave-variations/pull/361 has been merged for `Nightly` & `BETA`, we'll need to create a new PR to enable the above on 15% on `Release` once `1.46.x` is released and slowly phase it to 100% while keeping an eye on Community/Socials for possible regressions/issues.

Asking some questions via https://bravesoftware.slack.com/archives/C01LKMP6X36/p1668103608663869 so we can get https://github.com/brave/brave-variations/pull/361 merged.

**Comment by LaurenWags** [comment]
Per discussion in https://bravesoftware.slack.com/archives/C01LKMP6X36/p1668727124420119?thread_ts=1668103608.663869&cid=C01LKMP6X36, have changed this to `QA/No`. Additional testing to be done when related griffin PRs are ready.
